### PR TITLE
refactor: de-react-ify

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-08-15).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Some components seem to be React when they don't have to be.

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

File: www/src/components/headCommon.astro [hunk 1]
```diff
@@ -14,25 +14,3 @@
     element.setAttribute("tabindex", "0");
   });
 </script>
-
```
Comment (juliusmarminge) [inline www/src/components/headCommon.astro:17]: when the themetoggler is no longer react we can move this part to the toggler itself which makes more sense imo

File: www/src/components/navigation/themeToggleButton.astro [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<div
+  class="flex bg-t3-purple-200/50 dark:bg-slate-700 p-2 w-fit mx-auto rounded-lg gap-3"
+>
+  <label class="cursor-pointer text-t3-purple-500 dark:text-slate-900">
```
Comment (juliusmarminge) [inline www/src/components/navigation/themeToggleButton.astro:4]: i made both icons the same color, can change the sun back to gold if you don't like it

Comment (nexxeln) [reply www/src/components/navigation/themeToggleButton.astro:4]: I like this more tbh

**Review by nexxeln — DISMISSED** [review]
Great stuff

File: www/src/components/navigation/themeToggleButton.astro (hunk 2 above)
Comment (juliusmarminge) [reply www/src/components/navigation/themeToggleButton.astro:4]: yea me too, keeping it then :)

**Comment by juliusmarminge** [comment]
should i try porting the TOC too? then it's only the typewriter left but i think that's fine unless we just go for a gif or smth? @nexxeln

**Comment by nexxeln** [comment]
> should i try porting the TOC too? then it's only the typewriter left but i think that's fine unless we just go for a gif or smth? @nexxeln

Yeah sure

**Comment by juliusmarminge** [comment]
@nexxeln I also made the change so that the H1 is "rendered" in the TOC without having to be explicitely defined as H1 and H2

**Pull request merged**